### PR TITLE
Load new D2B NeXus files

### LINKED ISSUE #20881: Allow LoadILLDiffraction to read calibrated data
NeXus files with calibrated data already applied will be produced for D2B (and eventually D20). This issue is to add the option to load the calibrated data.

### PR DESCRIPTION
**Pull request by ianbush**
The NeXus files for D2B have been updated* to provide calibrate data. This is stored in `data_scan/detector_data/data`, with the presence of the entry `data_scan/detector_data/raw_data` telling you the other data block is calibrated data.

Some scanned variables were also added with the same name, so the name when loaded into Mantid is now `name.property`.

*The NeXus files are not being used yet so they are still changed sometimes.

**To test:**

Check code and unit test logic. Try out the attached D2B and D20 files, check both load with no `DataType` given, the D2B loads for `Calibrated` or `Raw` and the D20 file loads for `Calibrated` and throws for `Raw`.

[example_data.zip](https://github.com/mantidproject/mantid/files/1440578/example_data.zip)

Fixes #20881.

**Release Notes** 

Updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/LoadILLDiffraction.cpp [hunk 1]
```diff
@@ -701,5 +724,20 @@ LoadILLDiffraction::getInstrumentFilePath(const std::string &instName) const {
...
+/**
+ * Returns true if the file contains calibrated data
+ *
+ * @param filename The filename to check
+ * @return True if the file contains calibrated data, false otherwise
+ */
+bool LoadILLDiffraction::containsCalibratedData(
+    const std::string &filename) const {
+  NexusDescriptor descriptor(filename);
+  // This is unintuitive, but if the file has calibrated data there are entries
```
Comment (gvardany) [inline Framework/DataHandling/src/LoadILLDiffraction.cpp]: :1st_place_medal:

**Review by gvardany — COMMENTED** [review]
Looks good, just a comment about the default.
Maybe we should also add a sample log about what it loaded, but I'd rather do this in my reduction branch, where I added many other logs.

File: Framework/DataHandling/src/LoadILLDiffraction.cpp [hunk 2]
```diff
@@ -93,6 +94,22 @@ void LoadILLDiffraction::init() {
   declareProperty(make_unique<WorkspaceProperty<MatrixWorkspace>>(
                       "OutputWorkspace", "", Direction::Output),
                   "The output workspace.");
+  std::vector<std::string> calibrationOptions{"", "Raw", "Calibrated"};
```
Comment (gvardany) [inline Framework/DataHandling/src/LoadILLDiffraction.cpp]: So the default is "Raw"? Maybe we should have "Auto" as default, which will do what the empty string does now?

Comment (ianbush) [reply Framework/DataHandling/src/LoadILLDiffraction.cpp]: The default is the empty string. I agree calling this `Auto` would be a bit nicer - makes it clear you should check what is going to happen!

**Comment by ianbush** [comment]
Having a way to confirm whether the data is raw/calibrated is a good idea. The D2B files have a new log, `Detetor.calibration_file`. It is currently loaded in any case, but how about we set that to 'none' for loading raw data?

File: Framework/DataHandling/src/LoadILLDiffraction.cpp [hunk 3]
```diff
@@ -94,8 +94,8 @@ void LoadILLDiffraction::init() {
   declareProperty(make_unique<WorkspaceProperty<MatrixWorkspace>>(
                       "OutputWorkspace", "", Direction::Output),
                   "The output workspace.");
-  std::vector<std::string> calibrationOptions{"", "Raw", "Calibrated"};
-  declareProperty("DataType", "Raw",
+  std::vector<std::string> calibrationOptions{"Auto", "Raw", "Calibrated"};
+  declareProperty("DataType", "Auto",
                   boost::make_shared<StringListValidator>(calibrationOptions),
                   "Select the type of data, with or without calibration "
                   "already applied. If blank then the calibrated data is "
```
Comment (gvardany) [inline Framework/DataHandling/src/LoadILLDiffraction.cpp]: It is not blank anymore, maybe better to say `by default the calibrated data....`

File: Framework/DataHandling/src/LoadILLDiffraction.cpp [hunk 4]
```diff
@@ -226,6 +227,14 @@ void LoadILLDiffraction::loadMetaData() {
     m_loadHelper.addNexusFieldsToWsRun(nxHandle, m_outWorkspace->mutableRun());
     nxStat = NXclose(&nxHandle);
   }
+
+  if (mutableRun.hasProperty("Detector.calibration_file")) {
+    if (getPropertyValue("DataType") == "Raw") {
+      mutableRun.getProperty("Detector.calibration_file")->setValue("none");
+      g_log.error() << "Set to none!" << std::endl;
```
Comment (gvardany) [inline Framework/DataHandling/src/LoadILLDiffraction.cpp]: I hope this is a leftover?

File: Framework/DataHandling/src/LoadILLDiffraction.cpp (hunk 3 above)
Comment (ianbush) [reply Framework/DataHandling/src/LoadILLDiffraction.cpp]: This is now fixed, blank -> Auto.

File: Framework/DataHandling/src/LoadILLDiffraction.cpp (hunk 4 above)
Comment (ianbush) [reply Framework/DataHandling/src/LoadILLDiffraction.cpp]: Whoops, indeed it is!

**Review by gvardany — APPROVED** [review]
Should be good now.

**Pull request merged**